pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1147,159 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func getHostCpusNum() (int, error) {
+	// The `nproc` util relies on the /proc/sys file system, that is not properly accessible from a container.
+	// So, have to run the command with access to the host FS.
+	out, err := exec.Command("chroot", "/hostfs", "nproc", "--all").Output()
+	if err != nil {
+		return -1, err
+	}
+	// remove "\n"
+	out = out[:2]
+	maxHostCpus, _ := strconv.Atoi(string(out))
+	return maxHostCpus, nil
+}
```

Comment:
Do we even care about Xen support at the moment? If yes, do we have an example of disabling some part of the logic for Xen?
The feature has been implemented with KVM/QEMU stack in mind.

Reply:
> Do we even care about Xen support at the moment?

We still try to support it. So it would be polite at least to have something in logs/comments.

> If yes, do we have an example of disabling some part of the logic for Xen? The feature has been implemented with KVM/QEMU stack in mind.

We have such checks for example here: https://github.com/lf-edge/eve/blob/01aef8f73f9421797a83009e471ca2ca44375d23/pkg/pillar/cmd/volumemgr/volumemgr.go#L189-L196

Reply:
Ok, thanks, I'll try to add the corresponding check.

Reply:
Done

Reply:
@giggsoff Didn't realize we had that code. I think we should have a hypervisor.UseVHost() or alternatively IsVHostSupport() call for this.

Reply:
Should we do something with this now?

Reply:
Let's clean this up later (together with the vhost code which uses same pattern).

Reply:
Ok. Won't do anything in the scope of this PR then.